pymeasure/instruments/hp/hp3478A.py

```diff
@@ -0,0 +1,491 @@
+            "Rext": {3E7: "R7","auto": "RA"},
+            }
+
+    TRIGGERS={
+            "auto": "T1",
+            "internal": "T1",
+            "external": "T2",
+            "single": "T3",
+            "hold": "T4",
+            "fast": "T5",
+            }
+
+    ERRORS={
+            1: "Calibration error",
+            2: "RAM error",
+            4: "ROM error",
+            8: "AD slope error",
+            16: "AD converter error",
+            32: "AD link error",
+           }
+
+    active_connectors = Instrument.measurement(
+        "B",
+        """Return selected connectors ("front"/"back"), based on front-panel selector switch
+        """,
+        get_process = (lambda x: get_status(x.encode(encoding="ASCII"), "front_rear")),
+        values={"back":0, "front":1},
+        map_values = True,
+        )
+
+    auto_range = Instrument.measurement(
+        "B",
+        """ Return auto-ranging status, returns False if manual range and True if auto-range active.
+        For manual range control the range property can be set
+        """,
+        get_process = (lambda x: get_status(x.encode(encoding="ASCII"), "auto_range")),
+        values={False:0, True:1},
+        map_values = True,
+        )
+
+    auto_zero = Instrument.control(
+        "B",
+        "Z%d",
+        """ Returns autozero settings on the HP3478, this property can be set 
+        (False: disabled, True: enabled) 
+        """,
+        get_process = (lambda x: get_status(x.encode(encoding="ASCII"), "auto_zero")),
+        validator = strict_discrete_set,
+        values={False:0, True:1},
+        map_values = True,
+        )
+
+    cal_enable = Instrument.measurement(
+        "B",
+        """Return calibration enable switch setting (False: cal disabled, True: cablibration possible),
+        based on front-panel selector switch
+        """,
+        get_process = (lambda x: get_status(x.encode(encoding="ASCII"), "cal_enable")),
+        values={False:0, True:1},
+        map_values = True,
+        )
+
+    def check_errors(self):
+        """
+        Method to read the error status register
+
+        :return error_status: one byte with the error status register content
+        :rtype error_status: int
+        """
+
+        if self.error_status != 0:
+            log.critical("HP3478A error detected: $s", self.ERRORS[self.error_status] )
+        return self.error_status
+
+    error_status = Instrument.measurement(
+        "E",
+        """Checks the error status register
+        """,
+        get_process = (lambda x: int(x)),
+        )
+
+    disp_reset = Instrument.setting(
+        "D%d",
+        """Resets the display of the instrument if 1 is written to it.
+        """,
+        set_process = (lambda x: int(x)),
+        validator = strict_discrete_set,
+        values = [1],
+        )
+
+    #TODO: find a proper validator for the next two methods (without using regex, if possible)
```

Comment:
What kind of validator do you want to implement (in a broad sense, so what is it that you want to check)

Reply:
The insturment allows for text to be shown on the display only tall ASCII-letter, numbers, colon and dot.
My Idea was to have some kind of validator to check for this.
The set process with  str.upper() is already taking care of this, so I willr emove this comment for the time being.

Reply:
Check, perfect